File: app/src/main/java/com/snaptiongame/app/data/models/FitBCaption.java
package com.snaptiongame.app.data.models;

import com.google.gson.annotations.SerializedName;


/**
 * @author Nick Romero
 */

public class FitBCaption {
    @SerializedName(ID)
    public int id;
    @SerializedName(BEFORE)
    public String beforeBlank;
    @SerializedName(AFTER)
    public String afterBlank;
    @SerializedName(CHAR_LIMIT)
    public int charLimit;
    @SerializedName(SET_ID)
    public int setId;
    @SerializedName(USER_ENTRY)
    public String userEntry;

    public static final String ID = "id";
    public static final String BEFORE = "before";
    public static final String AFTER = "after";
    public static final String CHAR_LIMIT = "charLimit";
    public static final String SET_ID = "set_id";
    public static final String USER_ENTRY = "userEntry";
    public final String placeholderText = "______";

    public FitBCaption(int id, int charLimit, String beforeBlank, String afterBlank, int setId) {
        this.id = id;
        this.beforeBlank = beforeBlank;
        this.afterBlank = afterBlank;
        this.charLimit = charLimit;
        this.setId = setId;
        this.userEntry = "";
    }
    @Override
    public boolean equals(Object obj) {
        return obj instanceof FitBCaption && ((FitBCaption) obj).id == id;
    }
}

File: app/src/main/java/com/snaptiongame/app/data/models/FitBCaption.kt
package com.snaptiongame.app.data.models

import com.google.gson.annotations.SerializedName


/**
 * @author Nick Romero
 */

class FitBCaption(@SerializedName(ID) val id: Int,
                  @SerializedName(CHAR_LIMIT) val charLimit: Int,
                  @SerializedName(BEFORE) val beforeBlank: String,
                  @SerializedName(AFTER) val afterBlank: String,
                  @SerializedName(SET_ID) val setId: Int) {

    override fun equals(other: Any?): Boolean {
        return other is FitBCaption && other.id == id
    }

    companion object {
        const val ID = "id"
        const val BEFORE = "before"
        const val AFTER = "after"
        const val CHAR_LIMIT = "charLimit"
        const val SET_ID = "set_id"
        const val PLACEHOLDER = "______"
    }
}
